source/developers/transform-data-with-object-lambda.rst

```diff
@@ -0,0 +1,242 @@
+
+.. code-block:: shell
+   :class: copyable
+
+   pip install flask requests
+
+The handler uses information from the MinIO event context to get the original object, along with the tokens needed to validate the response when the transformed object is returned.
+In the example below, ``object_context`` contains several values from the ``getObjectContext`` property of the JSON request payload:
+
+* ``inputS3Url`` is a presigned URL for the original object. A `presigned URL <https://min.io/docs/minio/linux/developers/go/API.html#presigned-operations>`__ allows the function to access an object without the MinIO credentials usually required. 
+* ``outputRoute`` and ``outputToken`` are tokens added to the response headers to validate the response and return it to the correct user.
+
+The handler calls string ``upper()`` to convert the text to all uppercase letters.
+It then sends the results back to MinIO, which returns it to the caller.
+The original object is not changed.
+
+.. code-block:: py
+   :class: copyable
+
+   from flask import Flask, request, abort, make_response
+   import requests
+
+   app = Flask(__name__)
+   @app.route('/', methods=['POST'])
+   def get_webhook():
+      if request.method == 'POST':
+         # obtain the request event from the 'POST' call
+         event = request.json
+
+         object_context = event["getObjectContext"]
+
+         # Get the presigned URL to fetch the requested
+         # original object from MinIO
+         s3_url = object_context["inputS3Url"]
+
+         # Extract the route and request token from the input context
+         request_route = object_context["outputRoute"]
+         request_token = object_context["outputToken"]
+
+         # Get the original S3 object using the presigned URL
+         r = requests.get(s3_url)
+         original_object = r.content.decode('utf-8')
+
+         # Transform all text in the original object to uppercase
+         # You can replace it with your custom code based on your use case
+         transformed_object = original_object.upper()
+
+         # Write object back to S3 Object Lambda
+         # response sends the transformed data
+         # back to MinIO and then to the user
+         resp = make_response(transformed_object, 200)
+         resp.headers['x-amz-request-route'] = request_route
+         resp.headers['x-amz-request-token'] = request_token
+         return resp
+
+      else:
+         abort(400)
+
+   if __name__ == '__main__':
+      app.run()
+
+You may perform any actions needed for your use case, including saving the transformed object.
+See the :doc:`The MinIO Python SDK </developers/python/minio-py>` for more about working with objects programmatically.
+
+
+Start the Handler
++++++++++++++++++
+
+To test the handler in your local development environment, start it with the following command:
+
+.. code-block:: shell
+   :class: copyable
+
+   python lambda_handler.py
```

Comment:
ehhh for me it's `python3` but I think that's sufficiently well understood (by Python people) that I'm going to gloss over it. (Same for venv.)

Reply:
Maybe
> Use the following command to start the handler for testing in your local development environment:

Reply:
Also, yeah, just `python` is fine.

Reply:
I simplified the sentence and bypassed the awkward wording entirely. Good idea.